src/base/util_func.h

```diff
@@ -73,4 +77,12 @@ extern int util_log_write_command (int argc, char *argv[]);
 extern int util_bsearch (const void *key, const void *base, int n_elems, unsigned int sizeof_elem,
 			 int (*func_compare) (const void *, const void *), bool * out_found);
 
+// *INDENT-OFF*
+#if defined __cplusplus
+template <typename Duration>
+void util_get_seconds_and_rest_since_epoch (std::chrono::seconds &secs, Duration &rest);
+void util_get_second_and_ms_since_epoch (time_t &secs, int &msec);
```

Comment:
`util_get_second_and_ms_since_epoch` function should not be guarded by `__cplusplus`, I think.
`util_get_seconds_and_rest_since_epoch` maybe should not be be exposed at all since not used externally. And the header to remain pure C?

Reply:
To make it pure C, util_get_second_and_ms_since_epoch signature should be changed to `(time_t * secs, int * msec)`. Should be ok though :)

Reply:
done